sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/SclService.java

```diff
@@ -447,46 +459,55 @@ private static Map<String, TPrivate> createMapIEDNameAndPrivate(SclRootAdapter s
+                                && getCompasICDHeader(tPrivate).isPresent() && getCompasICDHeader(tPrivate).get().getIEDName() != null)
+                .collect(Collectors.toMap(tPrivate -> getCompasICDHeader(tPrivate).get().getIEDName(), tPrivate -> tPrivate));
     }
 
     private static Map<String, Pair<TPrivate, List<SCL>>> createMapICDSystemVersionUuidAndSTDFile(Set<SCL> stds) {
         Map<String, Pair<TPrivate, List<SCL>>> stringSCLMap = new HashMap<>();
         stds.forEach(std -> std.getIED().forEach(ied -> ied.getPrivate().forEach(tp -> {
-            if (tp.getType().equals(CommonConstants.COMPAS_ICDHEADER)) {
-               String icdSysVer = getValueFromPrivate(tp, CommonConstants.ICD_SYSTEM_VERSION_UUID);
-               if(icdSysVer != null) {
-                   Pair<TPrivate, List<SCL>> pair = stringSCLMap.get(icdSysVer);
-                   List<SCL> list = pair != null ? pair.getRight() : new ArrayList<>();
-                   list.add(std);
-                   stringSCLMap.put(icdSysVer, Pair.of(tp,list));
-               }
+            Optional<TCompasICDHeader> optIcdHeader = getCompasICDHeader(tp);
+            if (optIcdHeader.isPresent() && optIcdHeader.get().getICDSystemVersionUUID() != null) {
+                String icdSysVer = optIcdHeader.get().getICDSystemVersionUUID();
+                Pair<TPrivate, List<SCL>> pair = stringSCLMap.get(icdSysVer);
+                List<SCL> list = pair != null ? pair.getRight() : new ArrayList<>();
+                list.add(std);
+                stringSCLMap.put(icdSysVer, Pair.of(tp, list));
             }
         })));
         return stringSCLMap;
     }
 
-    private static String getValueFromPrivate(TPrivate tPrivate, String attributName) {
-        return tPrivate.getContent().stream()
-                .filter(o -> !o.toString().trim().isBlank())
-                .findFirst()
-                .map(o -> ((Element) o).getAttribute(attributName))
-                .orElse(null);
-    }
-
-    private static void checkSTDPrivateAndLNodePrivate(TPrivate iedPrivate, TPrivate scdPrivate) throws ScdException {
-        List<String> attributNames = Arrays.asList(CommonConstants.IED_TYPE, CommonConstants.ICD_SYSTEM_VERSION_UUID,
-                CommonConstants.VENDOR_NAME, CommonConstants.IED_REDUNDANCY, CommonConstants.IED_MODEL,
-                CommonConstants.HW_REV, CommonConstants.SW_REV, CommonConstants.HEADER_ID, CommonConstants.HEADER_VERSION,
-                CommonConstants.HEADER_REVISION);
-
-        Element iedElement = (Element) iedPrivate.getContent().get(1);
-        Element scdElement = (Element) scdPrivate.getContent().get(1);
-        boolean isEq = attributNames.stream().map(s -> iedElement.getAttribute(s).equals(scdElement.getAttribute(s))).reduce(true, (a, b) -> a && b);
-        if(isEq){
+    private static void compareAndCopyLNodePrivateIntoSTDPrivate(TPrivate iedPrivate, TPrivate scdPrivate) throws ScdException {
```

Comment:
Not 'clean code' to have 2 responsibilities. This method should be split in two : 
```java
if iedMatchScd(scdPrivate, iedPrivate){
copyCompasICDHeader(scdPrivate, iedPrivate)
}
```